pkg/osbuild/hmac_stage.go

```diff
@@ -0,0 +1,44 @@
+package osbuild
+
+import (
+	"fmt"
+	"slices"
+	"strings"
+)
+
+type HMACStageOptions struct {
+	Paths     []string `json:"paths"`
+	Algorithm string   `json:"algorithm"`
+}
+
+func (o *HMACStageOptions) isStageOptions() {}
+
+func (o *HMACStageOptions) validate() error {
+	if len(o.Paths) == 0 {
```

Comment:
This should check `o` for nil first, or change `NewHMACStage` to check options for nil.